instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/BeanPack.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine;
+
+import javax.management.ObjectName;
+import javax.management.QueryExp;
+
+/**
+ * A class describing a set of MBeans which can be used to collect values for a metric. Objects of
+ * this class are immutable.
+ */
+public class BeanPack {
+  // How to specify the MBean(s)
+  private final QueryExp queryExp;
```

Comment:
is non-null `queryExp` used?

Reply:
Currently it is not.
This part of the code would be activated by accessing JMX Metric Insight engine directly, without involving YAML parser. Examples of such use have been removed from this PR, just to minimize the size of the code to review.

It will be also possible to introduce the concept of _filters_ to the YAML schema, translating them to `QueryExp` objects.